wallet/src/repository/wallet/tests/mod.rs

```diff
@@ -586,6 +592,14 @@ fn test_create_vtt_does_not_spend_utxos() {
         })
         .unwrap();
 
+    // the extended transaction should actually contain a value transfer transaction
+    let vtt = if let Transaction::ValueTransfer(vtt) = extended.transaction {
+        Some(vtt)
+    } else {
+        None
+    }
+    .unwrap();
```

Comment:
Using `None.unwrap()` will not show us why this test failed, prefer a panic like this:

```suggestion
    let vtt = if let Transaction::ValueTransfer(vtt) = extended.transaction {
        vtt
    } else {
        panic!("the extended transaction should actually contain a value transfer transaction, got: {:?}", extended.transaction);
    };
```

If that doesn't compile, try using a match instead of if let.

Reply:
Agreed and fixed!